testkit/src/api.rs

```diff
@@ -280,4 +279,58 @@ impl TestKitApi {
             false,
         )
     }
+
+    /// Converts iron Response to ApiError.
+    ///
+    /// # Panics
+    ///
+    /// - Panics if the response has a non-error response status.
+    fn response_to_api_error(response: Response) -> ApiError {
+        let status = response.status.expect("Status header is not set");
+
+        fn error(response: Response) -> String {
+            let body = response::extract_body_to_string(response);
+            serde_json::from_str::<JsonValue>(&body)
+                .map(|value| match value {
+                    JsonValue::Object(object) => {
+                        object
+                            .get("description")
+                            .map(|value| value.as_str().map(String::from))
+                            .unwrap_or_else(|| serde_json::to_string(&object).ok())
+                            .unwrap_or_default()
+                    }
+                    JsonValue::String(string) => string,
```

Comment:
Are there tests that test various parsing possibilities:

- [ ] Object with `description` field
- [x] Object without `description` field
- [ ] JSON string
- [ ] Valid JSON, but not a string or an object
- [x] non-JSON input

I've marked those that seem to be implemented right now (maybe, I've missed some).